packages/actions/src/uploadSplitDocumentation/index.ts

```diff
@@ -20,13 +21,31 @@ for await (const file of globber.globGenerator()) {
 	try {
 		promises.push(
 			// eslint-disable-next-line @typescript-eslint/no-loop-func
-			limit(async () => {
+			queue.add(async () => {
 				console.log(`Uploading ${file} with ${version} from ${pkgName}...`);
 				const name = basename(file).replace('main.', '');
-				await put(`rewrite/${pkgName}/${version}.${name}`, data, {
-					access: 'public',
-					addRandomSuffix: false,
-				});
+				async function upload(retries = 0) {
+					try {
+						await put(`rewrite/${pkgName}/${version}.${name}`, data, {
+							access: 'public',
+							addRandomSuffix: false,
+						});
+					} catch (error) {
+						if (retries > 3) {
+							console.error(`Could not upload ${file} after 3 retries`, error);
+							return;
+						}
+
+						if (typeof error === 'object' && error && 'retryAfter' in error && typeof error.retryAfter === 'number') {
+							await sleep(error.retryAfter * 1_000);
+							return upload(retries + 1);
+						} else {
+							console.error(`Could not upload ${file}`, error);
+						}
+					}
+				}
```

Comment:
Could we do something like https://docs.github.com/en/actions/sharing-automations/creating-actions/setting-exit-codes-for-actions#setting-a-failure-exit-code-in-a-javascript-action here?

Reply:
```suggestion
					} catch (error) {
						if (retries > 3) {
							console.error(`Could not upload ${file} after 3 retries`, error);
							return;
						}

						if (typeof error === 'object' && error && 'retryAfter' in error && typeof error.retryAfter === 'number') {
							await sleep(error.retryAfter * 1_000);
							return upload(retries + 1);
						} else {
							console.error(`Could not upload ${file}`, error);
							process.exitCode = 1;
						}
					}
				}
```

Reply:
We already have `@actions/core` so we might as well use `core.setFailed`

Reply:
yeah that makes sense, we can set failed for either case here, good shout